core/embed/rust/src/trezorhal/uzlib.rs

```diff
@@ -0,0 +1,58 @@
+use super::ffi;
+use core::{marker::PhantomData, mem::MaybeUninit, ptr};
+
+pub const UZLIB_WINDOW_SIZE: usize = 1 << 10;
+pub use ffi::uzlib_uncomp;
+
+impl Default for ffi::uzlib_uncomp {
+    fn default() -> Self {
+        unsafe { MaybeUninit::<Self>::zeroed().assume_init() }
+    }
+}
+
+pub struct UzlibContext<'a> {
+    uncomp: ffi::uzlib_uncomp,
+    src_data: PhantomData<&'a [u8]>,
+    window: Option<[u8; UZLIB_WINDOW_SIZE]>,
+}
+
+impl<'a> UzlibContext<'a> {
+    pub fn new(src: &'a [u8], use_window: bool) -> Self {
+        let window = use_window.then(|| [0_u8; UZLIB_WINDOW_SIZE]);
```

Comment:
clippy complains that you should use `use_window.then_some([0u8...])`
i'm not entirely convinced due to stack usage but the Option is pre-sized anyway so 🤷‍♀️

Reply:
what clippy version do you use? i don't see it with 0.1.60

Reply:
fixed https://github.com/trezor/trezor-firmware/pull/2339/commits/7d89ad6b173b4850e9871574132886ee2807c88f

Reply:
hmm except our Rust does not know this :( https://gitlab.com/satoshilabs/trezor/trezor-firmware/-/jobs/2849519138
i'll go update Rust to latest stable, that should fix the problem

Reply:
no, wait, except it's supposed to be stable for like 8 versions already o_O